### ISSUE TITLE
drop cages?

### ISSUE THREAD

**Issue by nsteinme** [cortex-lab/alyx#147]
Fairly radical idea here: maybe we should drop the cages table altogether? The main point is that I don't think we really need the information for anything. It's not tracked at all currently in any of the sheets, in any case. Requiring people to deal with entering in things that they don't perceive as useful is likely to impair adoption. 

The one place where the concept of the cage is really useful in a practical sense is for keeping track of breeding "pairs" (which btw in practice sometimes have one male and two females). But then we should just have a table of breeding pairs, which have foreign keys to any number of subjects who are "parents" and to any number of litters that they produced. I'm not sure whether the litter having a mother and father, versus the litter having a "breeding group" (pair or triplet) is a clearer way to represent the lineage. Probably either way is fine.

What we'd lose is knowing which mice if any are co-housed. Like I mentioned, this information isn't currently recorded anywhere. On the other hand you could imagine being interested to know, e.g. you might be interested to know whether co-house mice do better at behavioral tasks. 

Another thing we'd lose is a way to represent where the mouse is housed. E.g. a cage has a physical location but a subject doesn't, so without cages there's no way to indicate which room the mouse lives in. In practice, at least right now, this is never ambiguous anyway, so again this information isn't recorded anywhere currently. 

Another solution, instead of getting rid of cages, would be to have cages, but just make it clear that people shouldn't feel like they need to enter anything about them or have a cage or anything. They could make a cage to put both their mice in if they want to make it clear that they are co-housed, but otherwise don't bother with it. (But, is it possible to reconstruct when the mouse was put into one cage and when into another? I think it's not, which means if we really want a record of the co-housing then we need a table like "water restrictions" where you record the start and end dates of the co-housing, and which subjects were in it). 

Thoughts? 
@kdharris101 @nippoo @k1o0 (I will run this by Laura and/or Charu too)

**Comment by rossant** [comment]
I like the last proposition. We don't change anything and we let the users interact with the software. Usage will tell if this feature is frequently used or not...

**Comment by nsteinme** [comment]
Ok, yes that sounds reasonable. I kind of like having a "breeding pair" table rather than cages that have multiple mice in them, but it's probably equivalent. We can keep it in mind, anyway.

**Comment by kdharris101** [comment]
Ø  to have cages, but just make it clear that people shouldn't feel like they need to enter anything about them or have a cage or anything

I think this is a bad idea. Having things that are filled out only half the time is a recipe for trouble – because once people stop bothering with one thing, they will stop bothering with lots of others too.

I think dropping it is OK. The only problem is if you want to know where to find a mouse, how do you find that? One option is to have a “current home cage” field in the subjects table. We won’t get the history, but that might not matter.

**Comment by nsteinme** [comment]
"Current home cage" is exactly the thing we already have. As you say,
there's no history associated with it.

The good thing is I don't think there's anything besides cage that people
**can** skip filling out. Everything else is already there.

Currently there is just no ambiguity about where mice live. If it's been
taken for a procedure it lives in 4.03, and if it's in the stock it's in
4.09. That's it. This is why it's kind of "meaningless information" - the
cage, and the cage's location - that we'd rather not burden people with.

What do you think about having a "breeding pair" table that links to
litters and parents? Then this doesn't show up for the Subject at all,
except via the subject's litter.

**Comment by kdharris101** [comment]
I don’t see why we need a breeding pair table – can’t the litter table have an entry for mother and father?

**Comment by nsteinme** [comment]
Let's discuss this next time you're in lab?

**Comment by nippoo** [comment]
We decided (several months ago) to drop the 'cages' table altogether and replace it with a "LAMIS mouse ID/reference" (as it happens, the same barcode as on the cage card) and let LAMIS deal with all the cage stuff if people wanted needed to know that info.

This probably got reversed at some point over the last couple of weeks but I'm still in favour of that original decision...

**Comment by nsteinme** [comment]
Ok, I never heard about that decision, sorry. Let me be clear about the point of my thinking. The one thing that really must be true is that entering new subjects must be as easy or easier as it currently is in the spreadsheets - _and also_ we need to be providing more information than we provided before, namely we want to record the parents of the new subject in a standardized way. 

I had imagined that we would represent reality in the database by first creating a cage and putting each parent into the cage, then creating a litter _from the cage_, such that the litter gets the correct parents and line (from the cage). I was struggling with this because it's not easier than the current system. However, perhaps the right answer is to skip the cage entirely and create a litter directly, specifying the parents and line at the time of litter creation. This doesn't really mirror reality since the litter comes out of nowhere, but it may make things as fast and easy as the spreadsheet.

Let me talk with Laura tomorrow to get more detail on exactly what she currently does and I'll try out this new way of creating new subject entries. 

Max, why wasn't the LAMIS thing implemented? In that case, the Cage table should have been dropped and the "Cage" field of the subject should switch to an integer rather than a foreign-key (LAMIS cage numbers are integers).
